src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,307 @@
+package parser;
+
+import command.*;
+import exception.TobiasException;
+import task.TaskList;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    
+    /**
+     * Takes in a string representation of date/time and returns it as a LocalDateTime object.
+     *
+     * @param dateTime String of the required date/time information.
+     * @return a LocalDateTime object with the provided date/time information.
+     * @throws TobiasException if the dateTime string does not follow format of dd-MM-yyyy HHmm.
+     * */
+    public static LocalDateTime dateFromString(String dateTime) throws TobiasException {
+        try {
+            return LocalDateTime.parse(dateTime.trim(), DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"));
+        } catch (Exception e) {
+            throw new TobiasException("Kindly enter the date or time in this format : dd-MM-yyyy HHmm");
+        }
+    }
+
+    /**
+     * Takes in a command as a String and a TaskList.
+     * Interprets that command and returns the relevant Command object.
+     *
+     * @param command Type of command given as a String
+     * @param tasks TaskList
+     * @throws TobiasException for each possible type of commands and/or if command is not valid.
+     * @return The relevant type of Command object.
+     * */
+    public static Command parseCommands(String command, TaskList tasks) throws TobiasException {
+        if (command.equals("hello")) {
+            return new ReplyCommand("Howdy there partner !");
+        } else if(command.equals("list")) {
+            return new ListCommand();
+        } else if (command.startsWith("todo")) {
+            if (command.equals("todo")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(4,5);
```

Comment:
weak SLAP, you should isolate the command processing (especially finding substrings) from the matching of commands